jruby/neo4j/driver/ext/config_converter.rb

```diff
@@ -70,7 +70,7 @@ def notification_config(minimum_severity: nil, disabled_categories: nil)
           org.neo4j.driver.internal.InternalNotificationConfig.new(
             value_of(org.neo4j.driver.internal.InternalNotificationSeverity, minimum_severity),
             disabled_categories
-              &.map { |value| value_of(org.neo4j.driver.internal.InternalNotificationCategory, value) }
+              &.map { |value| org.neo4j.driver.NotificationCategory.const_get(value.to_s.upcase) }
```

Comment:
We have been dealing with this issue the following way: https://github.com/neo4jrb/neo4j-ruby-driver/blob/5.0/ruby/neo4j/driver/access_mode.rb
to have some level of stronger typing compatible with java.
I could be swayed over but could end up with complex converting code when the configuration is deeply nested resulting in runtime errors as users would be using nested symbol hashes instead of higher level types.

Reply:
Using NotificationClassification serves as well as a type of documentation.